docs_src/V2TopLevelMigration.md

```diff
@@ -0,0 +1,165 @@
+# V2 Migration Guide
+
+!!! edgey "EdgeX 2.0"
+    Many backward breaking changes occurred in the EdgeX 2.0 (Ireland) release which may require some migration depending on your use case.
+
+This section describes how to migrate from EdgeX 1.x to EdgeX 2.0 at a high level and refers the reader to the appropriate detail documents. The areas to consider for migrating are:
+
+- [Custom Compose File](#custom-compose-file)
+- [Database](#database)
+- [Custom Configuration](#custom-configuration)
+- [Custom Device Service](#custom-device-service)
+- [Custom Device Profile](#custom-device-profile)
+- [Custom Pre-Defined Device](#custom-pre-defined-device)
+- [Custom Applications Service](#custom-applications-service)
+- [Security](#security)
+
+## Custom Compose File
+
+The compose files for V2 have many changes from their V1 counter parts. If you have customized a V1 compose file to add additional services or tweak overrides,  it is highly recommended that you start with the appropriate V2 compose file and re-add your customizations. It is very likely that the sections for your additional services will need to be migrated to have the proper environment overrides. Best approach is to use one of the V2 service sections that closest matches your service  as a template.
+
+The latest V2 compose files can be found here: [https://github.com/edgexfoundry/edgex-compose/tree/ireland](https://github.com/edgexfoundry/edgex-compose/tree/ireland)
+
+### Compose Builder
+
+If the add on service(s) in your custom compose file are EdgeX released device or app services, it is highly recommended that you use the Compose Builder to generate your custom compose file. 
+
+The latest V2 Compose Builder can be found here: [https://github.com/edgexfoundry/edgex-compose/tree/ireland/compose-builder#readme](https://github.com/edgexfoundry/edgex-compose/tree/ireland/compose-builder#readme)
+
+## Database
+
+There currently is no migration path for the data stored in the database. The V2 data collections are stored separately from the V1 data collections in the Redis database. Redis is now the only supported database, i.e. support for Mongo has been removed.
+
+!!! note
+    Since the V1 data and V2 data are stored separately, one could create a migration tool and upstream it to the EdgeX community.
+
+!!! warning
+    If the database is not cleared before starting the V2 services, the old V1 data will still reside in the database taking up useful memory. It is recommended that you first wipe the database clean before starting EdgeX 2.0 Services. That is unless you create a DB migration tool. See [Clearing Redis Database](#clearing-redis-database) section below for details on how to clear the Redis database.
```

Comment:
I would leave the previous note, but then remove the sentence "This is unless you create a DB migration tool."

Reply:
The point I am making is you don't want to clear the V1 DB if you have created a migration tool.

Reply:
Added more detail to that comment.